crypto/bn/bn_rsa_fips186_4.c

```diff
@@ -329,7 +329,7 @@ int ossl_bn_rsa_fips186_4_derive_prime(BIGNUM *Y, BIGNUM *X, const BIGNUM *Xin,
                     || !BN_sub_word(y1, 1)
                     || !BN_gcd(tmp, y1, e, ctx))
                 goto err;
-            if (BN_is_one(tmp) && BN_check_prime(Y, ctx, cb))
+            if (BN_is_one(tmp) && BN_check_prime(Y, ctx, cb) > 0)
```

Comment:
You could also "goto err" in case it returns -1

Reply:
Done. Thanks for your reminder.